caffe2/utils/threadpool/pthreadpool-cpp.cc

```diff
@@ -80,8 +80,8 @@ PThreadPool* pthreadpool() {
     pthread_atfork(nullptr, nullptr, child_atfork);
   });
 #endif
-  auto true_bool = true;
-  if (leak_corrupted_threadpool.compare_exchange_strong(true_bool, false)) {
+  if C10_LIKELY(leak_corrupted_threadpool) {
```

Comment:
Is `C10_LIKELY` used for readability? I can remove it, if required.

Reply:
Shouldn't this be `C10_UNLIKELY`? In any case, this can help the branch predictor but it feels like you've got it inverted here. Also minor nit, enclose it in parentheses.

Reply:
It should be `C10_LIKELY` because this condition should be entered when `leak_corrupted_threadpool` is `true`.


> Also minor nit, enclose it in parentheses.
This comment suggests I shouldn't :smiley:, but I can do so nevertheless -
https://github.com/pytorch/pytorch/blob/b39eeb07ed2f60a27b60ad32d172b58d31554282/c10/macros/Macros.h#L159-L165

Reply:
@ezyang, it does seem like `C10_LIKELY` and `C10_UNLIKELY` are now used inside parentheses, as in #55553 that was opened today, so I followed through on your advice. Thanks!

Reply:
> It should be C10_LIKELY because this condition should be entered when leak_corrupted_threadpool is true.

`C10_LIKELY(expr)` means that `expr` is likely to be true. I think most of the time leak_corrupted_threadpool will be false, right?

Reply:
Oh, I see. I was confused by the following line:

https://github.com/pytorch/pytorch/blob/b39eeb07ed2f60a27b60ad32d172b58d31554282/c10/macros/Macros.h#L175

I just looked up what [`__builtin_expect`](https://gcc.gnu.org/onlinedocs/gcc/Other-Builtins.html) means, so I should indeed use `C10_UNLIKELY`:
https://github.com/pytorch/pytorch/blob/b39eeb07ed2f60a27b60ad32d172b58d31554282/c10/macros/Macros.h#L173

Thanks a lot for your help! I'll fix it asap.